From 43f218522128b7864346bb11f7aad234410db745 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 8 Dec 2011 15:04:04 +0100
Subject: [PATCH] 95iscsi/iscsiroot: unset used variables before starting

If iscsiroot is called multiple times, then some variables can hold the
values of a previous call, so unset all variables before using them.

https://bugzilla.redhat.com/show_bug.cgi?id=752066
---
 modules.d/95iscsi/iscsiroot |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/modules.d/95iscsi/iscsiroot b/modules.d/95iscsi/iscsiroot
index bcdc046..e7bac74 100755
--- a/modules.d/95iscsi/iscsiroot
+++ b/modules.d/95iscsi/iscsiroot
@@ -51,6 +51,11 @@ if getargbool 0 rd.iscsi.firmware -y iscsi_firmware ; then
     exit 0
 fi
 
+unset iscsi_initiator iscsi_target_name iscsi_target_ip iscsi_target_port
+unset iscsi_target_group iscsi_protocol iscsirw iscsi_lun
+unset iscsi_username iscsi_password 
+unset iscsi_in_username iscsi_in_password
+
 # override conf settings by command line options
 arg=$(getargs rd.iscsi.initiator iscsi_initiator=)
 [ -n "$arg" ] && iscsi_initiator=$arg
